src/bio/bio_monitor.c

```diff
@@ -623,6 +625,18 @@ bio_init_health_monitoring(struct bio_blobstore *bb, char *bdev_name)
 	D_ASSERT(channel != NULL);
 	bb->bb_dev_health.bdh_io_channel = channel;
 
+	/** register DAOS sensors to export NVMe stats */
+#define X(field, fname, desc, unit)					\
+	rc = d_tm_add_metric(&bb->bb_dev_health.field, D_TM_COUNTER,	\
+			     desc, unit, "/nvme/%s/%s", bdev_name,	\
```

Comment:
Is there a shorter identifier for the bdev? Using bdev_name results in daos_metrics output like this:
```
                    nvme
                                        Nvme_wolf-61.wolf.hpdd.intel.com_0n1
                                                            commands
                                                                                data_units_written: 18472 512b data units
                                                                                data_units_read: 28461 512b data units
                                                                                host_write_cmds: 1306436 commands
                                                                                host_read_cmds: 194480 commands
                                                                                ctrl_busy_time: 0 minutes
                                                                                media_errs: 0
                                                                                read_errs: 0 errors
                                                                                write_errs: 0 errors
                                                                                unmap_errs: 0
                                                                                checksum_mismatch: 0 errors
```

In the prometheus exporter, we wind up with names like this:
```
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_checksum_mismatch Number of checksum mismatch detected by the engine
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_checksum_mismatch counter
engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_checksum_mismatch{rank="0"} 0
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_ctrl_busy_time Amount of time the controller is busy with I/O commands
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_ctrl_busy_time counter
engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_ctrl_busy_time{rank="0"} 0
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_read number of 512b data units read from to the controller
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_read counter
engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_read{rank="0"} 28461
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_written number of 512b data units written to the controller
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_written counter
engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_data_units_written{rank="0"} 18472
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_host_read_cmds number of read commands completed by to the controller
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_host_read_cmds counter
engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_host_read_cmds{rank="0"} 194480
# HELP engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_host_write_cmds number of write commands completed by to the controller
# TYPE engine_nvme_Nvme_wolf_61_wolf_hpdd_intel_com_0n1_commands_host_write_cmds counter
```

We have to do some processing in the exporter to convert the bdev identifier into a label anyhow, but using the full hostname seems a bit verbose...

Reply:
ah, i did not get this on my side on edv, interesting. @sydneyvanda any suggestion here?